File: culebrita/culebrita/Material(java)/P3C13Colas/src/main/java/com/cola/circular/ColaCircular.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.cola.circular;

/**
 *
 * @author Ruldin
 */
public class ColaCircular {
     protected int fin;
    private static int MAXTAMQ=40;
    protected int frente;
    
    protected Object[] listaCola;
    
    private int siguiente(int r){
        return (r+1) % MAXTAMQ;
    }
    
    
    public ColaCircular(){
        frente = 0;
        fin = MAXTAMQ-1;
        listaCola = new Object[MAXTAMQ];
    }
    
    
    /// paste
    // OPERACIONES DE MODIFICACIN DE LA COLA
    
    public void insertar(Object elemento) throws Exception{
        if (!colaLlena()){
            fin = siguiente(fin);
            listaCola[fin] = elemento;
        } else {
            throw new Exception("Overflow en la cola");
        }
        
    }
    
    
    public Object quitar() throws Exception {
        if (!colaVacia()){
            Object aux = listaCola[frente];
            frente = siguiente(frente);
            return aux;
        } else {
            throw new Exception("Cola vacia");
        }
    }
    
//cola vacia
public void borrarCola(){
    frente = 0;
    fin = MAXTAMQ-1;
}    
    
//acceso a la cola
public Object frenteCola() throws Exception{
    if (!colaVacia()){
        return listaCola[frente];
    } else {
        throw new Exception("Cola vacio");
    }
}
    

//metodos de verificacion de cola
public boolean colaVacia(){
    return frente == siguiente(fin);
}

public boolean colaLlena(){
    return fin == siguiente(siguiente(fin));
}

    
}

File: culebrita/culebrita/ColaCircular/ColaCircular.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LogicaColaCiruclar
{
    internal class ColaCircular
    {
        private static int fin;
        private static int MAXTAMQ = 900000000; //Tamaño maximo de la cola
        protected int frente;

        public int num;

        public int numeleme()//Metodo para obtener el numero de elementos de la cola
        {
            return num; 
        }



        protected Object[] listaCola;//Lista de elementos de la cola 

        private int siguiente(int r)//MEotod para obtener el siguiente elemento de la cola
        {
            return (r + 1) % MAXTAMQ;
        }

        public ColaCircular()//Constructor de la cola circular
        {
            frente = 0;
            fin = MAXTAMQ - 1;
            listaCola = new Object[MAXTAMQ];
        }

        public void insertar(Object elemento) //Metodo para insertar un elemento en la cola
        {
            if (!colaLlena())
            {
                num++;
                fin = siguiente(fin);//Obtiene el siguiente elemento de la cola
                listaCola[fin] = elemento;//Agrega el dato a la cola
            }
            else
            {
                throw new Exception("Overflow en la cola");
            }
        }//Fin del metodo insertar

        public Object quitar()//Metodo para quitar un elemento de la cola
        {
            if (!colaVacia())
            {
                num--;
                Object tm = listaCola[frente];//Obtiene el primer elemento de la cola
                frente = siguiente(frente);//Quita el primer elemento de la cola
                return tm;
            }
            else
            {
                throw new Exception("Cola vacia ");
            }
        }//Fin del metodo quitar

        public Object finalCola()//Metodo para obtener el ultimo elemento de la cola
        {
            if (!colaVacia())
            {
                return listaCola[fin];//Obtiene el ultimo elemento de la cola
            }
            else
            {
                throw new Exception("Cola vacia ");
            }
        }//Fin del metodo finalcola

        public bool colaVacia()
        {
            return frente == siguiente(fin);//Si la cola esta vacia
        }


        public bool colaLlena()
        {
            return frente == siguiente(siguiente(fin));//Si la cola esta llena
        }





    }

}
